src/psyclone/psyir/transformations/inline_trans.py

```diff
@@ -815,6 +771,54 @@ def _replace_formal_struc_arg(
         :returns: the replacement reference.
 
         '''
+        actual_arg = actual_arg.copy()
+
+        # If the local reference is a simple Reference then we can just
+        # replace it with a copy of the actual argument, e.g.
+        #
+        #   call my_sub(my_struc%data(i,j))
+        #
+        #   subroutine my_sub(var)
+        #     ...
+        #     var = 0.0
+        #
+        # pylint: disable=unidiomatic-typecheck
+        if type(ref) is Reference:
+            return actual_arg
+
+        # Below this point we need to know if the reference is to an Array
+        # and if so, what are their boundaries. To do so we can use the
+        # Reference2ArrayRangeTrans
+        if type(actual_arg) is Reference:
+            if isinstance(actual_arg.datatype, ArrayType):
+                dummy = Assignment()
+                dummy.addchild(actual_arg)
+                Reference2ArrayRangeTrans().apply(actual_arg)
+                actual_arg = dummy.children[0]
+
+        # Local reference is not simple but the actual argument is, e.g.:
+        #
+        #   call my_sub(my_struc)
+        #
+        #   subroutine my_sub(var)
+        #     ...
+        #     var%data(i,j) = 0.0
+        #
+        if type(actual_arg) is Reference:
```

Comment:
Perhaps it needs a new comment along the lines of "Actual argument is a scalar quantity"?